internal/k8s/controller.go

```diff
@@ -3980,7 +3980,7 @@ func (lbc *LoadBalancerController) HasCorrectIngressClass(obj interface{}) bool
 		class = obj.Annotations[ingressClassKey]
 		if class == "" && obj.Spec.IngressClassName != nil {
 			class = *obj.Spec.IngressClassName
-		} else {
+		} else if class != "" {
```

Comment:
@jjngx should there still be an ending`else {}` block according to go patterns?

Reply:
Not that I'm aware of.
This is the same as doing this:
```
class = obj.Annotations[ingressClassKey]
if class == "" && obj.Spec.IngressClassName != nil {
    class = *obj.Spec.IngressClassName
}

// True only if there is a value for obj.Annotations[ingressClassKey]
if class != "" {
    glog.Warningln("Using the DEPRECATED annotation 'kubernetes.io/ingress.class'. The 'ingressClassName' field will be ignored.")
}
```